client/webserver/site/src/js/mmsettings.ts

```diff
@@ -633,261 +1177,418 @@ export default class MarketMakerSettingsPage extends BasePage {
+      Doc.hide(page.profitSelectorBox)
+      this.setGapFactorLabels(page.gapStrategySelect.value || '')
     }
 
-    // Base wallet options
-    if (this.updatedConfig.basicMarketMakingConfig.baseOptions && this.originalConfig.basicMarketMakingConfig.baseOptions) {
-      for (const opt of Object.keys(this.updatedConfig.basicMarketMakingConfig.baseOptions)) {
-        const value = this.originalConfig.basicMarketMakingConfig.baseOptions[opt]
-        this.updatedConfig.basicMarketMakingConfig.baseOptions[opt] = value
-        if (this.baseWalletSettingControl[opt]) {
-          this.baseWalletSettingControl[opt].setValue(value)
-        }
+    // Buy/Sell placements
+    oldCfg.buyPlacements.forEach((p) => { this.addPlacement(true, p, viewOnly) })
+    oldCfg.sellPlacements.forEach((p) => { this.addPlacement(false, p, viewOnly) })
+
+    for (const opt of Object.keys(cfg.baseOptions)) {
+      const value = oldCfg.baseOptions[opt]
+      cfg.baseOptions[opt] = value
+      if (this.baseWalletSettingControl[opt]) {
+        this.baseWalletSettingControl[opt].setValue(value)
       }
     }
 
     // Quote wallet options
-    if (this.updatedConfig.basicMarketMakingConfig.quoteOptions && this.originalConfig.basicMarketMakingConfig.quoteOptions) {
-      for (const opt of Object.keys(this.updatedConfig.basicMarketMakingConfig.quoteOptions)) {
-        const value = this.originalConfig.basicMarketMakingConfig.quoteOptions[opt]
-        this.updatedConfig.basicMarketMakingConfig.quoteOptions[opt] = value
-        if (this.quoteWalletSettingControl[opt]) {
-          this.quoteWalletSettingControl[opt].setValue(value)
-        }
+    for (const opt of Object.keys(cfg.quoteOptions)) {
+      const value = oldCfg.quoteOptions[opt]
+      cfg.quoteOptions[opt] = value
+      if (this.quoteWalletSettingControl[opt]) {
+        this.quoteWalletSettingControl[opt].setValue(value)
       }
     }
 
     this.updateModifiedMarkers()
   }
 
+  /*
+   * validateFields validates configuration values and optionally shows error
+   * messages.
+   */
+  validateFields (showErrors: boolean): boolean {
+    let ok = true
+    const {
+      page, specs: { botType },
+      updatedConfig: { sellPlacements, buyPlacements, profit, useEmptyMarketRate, emptyMarketRate, baseBalance, quoteBalance }
+    } = this
+    const setError = (errEl: PageElement, errID: string) => {
+      ok = false
+      if (!showErrors) return
+      errEl.textContent = intl.prep(errID)
+      Doc.show(errEl)
+    }
+    if (showErrors) {
+      Doc.hide(
+        page.buyPlacementsErr, page.sellPlacementsErr, page.profitInputErr,
+        page.emptyMarketRateErr, page.baseBalanceErr, page.quoteBalanceErr
+      )
+    }
+    if (botType !== botTypeBasicArb) {
+      if (buyPlacements.length === 0) setError(page.buyPlacementsErr, intl.ID_NO_PLACEMENTS)
+      if (sellPlacements.length === 0) setError(page.sellPlacementsErr, intl.ID_NO_PLACEMENTS)
```

Comment:
Should we allow just placing on one side?

Reply:
Done